src/Rules/Pure/FunctionPurityCheck.php

```diff
@@ -86,6 +86,11 @@ public function check(
 				count($throwPoints) === 0
 				&& count($impurePoints) === 0
 				&& count($functionReflection->getAsserts()->getAll()) === 0
+				&& (
+					!$functionReflection instanceof ExtendedMethodReflection
```

Comment:
Do we need to support `@final` phpdoc?

Reply:
It would be better indeed.

It's not included in those isFinal methods ?

Reply:
I am not sure. A test could tell :-)

Reply:
tests added